judge/judgedaemon.main.php

```diff
@@ -872,6 +877,50 @@ function fetch_executable_internal(
         continue;
     }
 
+    if ($type == 'output_visualization') {
+        if ($lastWorkdir !== null) {
+            cleanup_judging($lastWorkdir);
+            $lastWorkdir = null;
+        }
+        foreach ($row as $judgeTask) {
+            if (isset($judgeTask['output_visualizer_script_id'])) {
+                // Visualization of output which this host judged requested.
+                $run_config = dj_json_decode($judgeTask['run_config']);
+                $tmpfile = tempnam(TMPDIR, 'output_visualization_');
+                [$runpath, $error] = fetch_executable_internal(
+                    $workdirpath,
+                    'output_visualization',
+                    $judgeTask['output_visualizer_script_id'],
+                    $run_config['hash']
+                );
+                if (isset($error)) {
+                    // FIXME
+                    continue;
+                }
+
+                $teamoutput = $workdir . "/testcase" . sprintf('%05d', $judgeTask['testcase_id']) . '/1/program.out';
```

Comment:
where does the `/1/` here come from?

Reply:
Getting the output of the first run, so in other words, this does not work for multipass problems.

Reply:
Ah, right - then please add a TODO and consider making the one a local variabled, e.g. `$pass = 1;` and then using that in the path construction.

Reply:
What would we want to do in this case? I assume we want to visualize actually the last pass?